### DISCUSSION TITLE
Error when trying to restart a print

### DISCUSSION THREAD

**Discussion by Identity-Unkn0wn** [jschuh/klipper-macros#114] [Q&A] [answered]
So when my ztilt fails and I press the reprint buttion I get this error. In fact the only was I can start another print sfter this is to restart Klipper all togehter. This issue only started after the last update that required me to change the start print gcode. Ive tried it on PrusaSlicer and SuperSlicer, seem to get the same error if z_tilt fails.

![klipper error](https://github.com/jschuh/klipper-macros/assets/55931640/bce2a6b7-9e57-4f32-9e09-10b29634a5bd)

**Comment by jschuh** [comment]
Are you printing from the virtual SD, and do you have your `on_error_gcode` set to `CANCEL_PRINT`? It would look something like this:

```
[virtual_sdcard]
path: ~/gcode_files
on_error_gcode: CANCEL_PRINT
```

**Reply by Identity-Unkn0wn** [reply]
Yes I am and my [virtual_sdcard] section looks like that.

Just a sn that I didn't mentiuon above, I followed the trouble shooting section and didnt get any errors when running the km check either

**Reply by jschuh** [reply]
That's very odd. It's clear from the error you're hitting that `CANCEL_PRINT` isn't running. But if you're printing from the SD card and you have `CANCEL_PRINT` in your `on_error_gcode`, [then it's going to run it](https://github.com/Klipper3d/klipper/blob/37315bf3365484c3ba6adef2b7da2b4464980e8a/klippy/extras/virtual_sdcard.py#L263-L271).

Anyway, I have a patch up on dev branch that I was already working on to fix some minor issues and weird cases for Octoprint. It should also make your case work, although it really shouldn't be necessary when `on_error_gcode` is configured.

**Reply by jschuh** [reply, marked as answer]
I pushed the changes from the dev branch to main this morning, so you can give it a try and see if it solves your issues.

**Reply by Identity-Unkn0wn** [reply]
This did the trick! Thanks!!!
